torch/csrc/jit/python/init.cpp

```diff
@@ -1398,79 +1597,132 @@ void initJITBindings(PyObject* module) {
               })
           .def(
               "__add__",
-              [](c10::SymIntNode a, py::object b) -> c10::SymIntNode {
-                auto snb = toSymIntNode(a, b);
-                return a->add(snb);
-              })
+              [](c10::SymIntNode a, py::object b) -> py::object {
+                if (is_int_or_symint(b)) {
+                  auto snb = toSymIntNode(a, b);
+                  return py::cast(a->add(snb));
+                } else {
+                  auto snb = toSymFloatNode(a, b);
+                  return py::cast(a->add(snb));
+                }
+              }, py::is_operator())
```

Comment:
From the pybind docs, I thought that adding this line would correctly tell pybind to catch `CastException` and return `NotImplemented`, instructing python to swizzle the arguments and try `__radd__`. That didn't actually work though. So I'll look around some more.